src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -258,7 +261,15 @@ func (c *controller) waitAndUpdateNodesStatus() bool {
 			}
 		}
 
-		if common.IsK8sNodeIsReady(node) && host.Host.Progress.CurrentStage != models.HostStageDone {
+		ready := common.IsK8sNodeIsReady(node)
+		if !ready && removeUninitializedTaint {
```

Comment:
It will not become ready till we will not remove it?

Reply:
It won't become ready with the taint (CNI won't schedule there)